### DISCUSSION TITLE
Any example about message schema?

### DISCUSSION THREAD

**Discussion by Seraphli** [foxglove/community#58] [Q&A]
I'm new to foxglove studio, and I notice I can visualize custom data with Foxglove schemas. Some of the messages are easy to understand. But some confused me.
Is there an example of how to write messages like Grid, which contain a binary data field? So do I have to serialize data myself, or is there any tool to serialize message structures from Foxglove schemas?

**Comment by jtbandes** [comment]
Sorry we don't have full example code right now, but you can see some of the data used in tests: https://github.com/foxglove/studio/blob/4f508a9f178e5e53f4d6f76fd4371df01a033eac/packages/studio-base/src/panels/ThreeDimensionalViz/index.stories.tsx#L1455-L1473

Grid in particular only supports Int8 values right now, so you just need your `data` to be a sequence of int8 bytes, and `fields` should contain a single entry with int8 type.

**Reply by Seraphli** [reply]
Thanks. This test script does provide an example. But I guess I will just use ros message now until there are some helper classes later.
